compendium/LogService/include/cppmicroservices/logservice/Logger.hpp

```diff
@@ -0,0 +1,110 @@
+#ifndef CPPMICROSERVICES_LOGGER_H__
+#define CPPMICROSERVICES_LOGGER_H__
+
+#include "cppmicroservices/ServiceReferenceBase.h"
+
+#include <cstdint>
+#include <exception>
+#include <string>
+#include <functional>
+
+namespace sinks
+{
+    class sink;
+}
+
+namespace spdlog
+{
+    class logger;
+    using sink_ptr = std::shared_ptr<sinks::sink>;
+} // namespace spdlog
+
+namespace cppmicroservices
+{
+    namespace logservice
+    {
+        /** @}*/
+
+        /**
+         * \ingroup MicroService
+         * \ingroup gr_logservice
+         *
+         * Provides methods for bundles to write messages to the log.
+         * Logger interface defines several methods for each of the defined LogLevels.
+         *
+         */
+
+	enum class LogLevel
+        {
+            audit, //This log level is used for information that must always be logged.
+ 
+            error, //This log level is used for information about an error situation.
+ 
+            warn, //This log level is used for information about a failure or unwanted situation that is not blocking.
+ 
+            info, //This log level is used for information about normal operation.
+ 
+            debug, //This log level is used for detailed output for debugging operations.
+ 
+            trace //This log level is used for large volume of output for tracing operations.   
+	};
+	
+        class Logger
+        {
+          public:
+
+            virtual ~Logger() = default;
+
+	    virtual void audit(const std::string& message) = 0;
+	    virtual void audit(const std::string& format, const std::string& arg) = 0;
+	    virtual void audit(const std::string& format, const std::string& arg1, const std::string& arg2) = 0;
+	    virtual void audit(std::string const& message, const std::exception_ptr ex) = 0;
```

Comment:
const in parameters are inconsistent across different interfaces. please follow one style of all interfaces